src/renderers/dom/shared/ReactDOMTextComponent.js

```diff
@@ -149,13 +150,29 @@ assign(ReactDOMTextComponent.prototype, {
     if (nativeNode) {
       return nativeNode;
     }
-    nativeNode = [this._openingComment, this._nativeNode];
+    if (!this._closingComment) {
```

Comment:
Woops, sorry, I guess I assumed useCreateElement=false was already tested. But this seems obvious now. Too bad we needed the code to find the other comment, after all.